db/migrate/20171026114327_add_deleted_on_to_container_quota_and_items.rb

```diff
@@ -1,9 +1,34 @@
 class AddDeletedOnToContainerQuotaAndItems < ActiveRecord::Migration[5.0]
+  class ContainerQuota < ActiveRecord::Base; end
+  class ContainerQuotaItem < ActiveRecord::Base
+    belongs_to :container_quota
+  end
+
   def change
     add_column :container_quotas, :deleted_on, :datetime
     add_index :container_quotas, :deleted_on
 
-    add_timestamps :container_quota_items
+    # Historical note: this migration has been edited. Original just naively added NOT
+    # NULL timestamp columns and failed with non-empty container_quota_items table.
+
+    add_timestamps :container_quota_items, :null => true # temporarily for backfilling, then disallow nulls
+    reversible do |dir|
+      dir.up do
+        say_with_time("Backfilling container_quota_items timestamps from parent quotas") do
+          # All quota items SHOULD have a parent quota, which SHOULD have deleted_on
+          # but just in case, fallback to current time.
+          now = Time.zone.now
+          ContainerQuotaItem.includes(:container_quota).find_each do |item|
```

Comment:
If you are using `.includes`, you have to be sure that the stub marks the belongs_to to use the right class name.